apm-agent-core/src/main/java/co/elastic/apm/agent/configuration/CoreConfiguration.java

```diff
@@ -337,7 +337,7 @@
         ));
 
     private final ConfigurationOption<List<MethodMatcher>> traceMethods = ConfigurationOption
-        .builder(new ListValueConverter<>(MethodMatcherValueConverter.INSTANCE), List.class)
+        .builder(new ListValueConverter<>(MethodMatcherValueConverter.INSTANCE, ",(?![^()]*\\))"), List.class)
```

Comment:
As there is only a single usage of this constructor here, we might better have a static factory method, maybe 
`methodListConverter()` that just returns `new ListValueConverter<>(MethodMatcherValueConverter.INSTANCE, ",(?![^()]*\\))")`. That would make code a bit more explicit, otherwise it feels a bit like a magic regex without much context, and the test duplicates that regex value (which does not guarantees that `traceMethods` fields will be properly working as expected.

Reply:
I was thinking that magic regex could be replaced by self explanatory name of constant (like `COMMA_OUT_OF_BACKETS`) and keep it only on one place (`ListValueConverter` could be good place).
But constructing method (like `methodListConverter()`) is not used with any other builder.
Where should it be?
Anyway `ListValueConverter` is also thread safe, there can be also single constant like
```java
enum MethodMatcherValueConverter ... {
 //...
    public static final ListValueConverter<MethodMatcher> LIST =
        new ListValueConverter<>(INSTANCE, COMMA_OUT_OF_BACKETS);
}
```
which can be simple used like:
```java
  MethodMatcherValueConverter.LIST
```
What you think?